### ISSUE TITLE
Use latest version of template

### ISSUE THREAD

**Issue by Dilshat** [subutai-io/agent#276]
We should always use latest version of template when importing it.

**Comment by Dilshat** [comment]
1. Check latest version on Kurjun
2. If local template is missing or has older version, download latest one.
    Remove installed template, and install new one
    Otherwise, no-op

**Comment by lbthomsen** [comment]
The changes made still doesn't work.

Steps taken.  In the debian-stretch template I have included a file /etc/subutai-template-version that contains the version:

```
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

Cleaned out everything (subutai destroy debian-stretch - removed all files from tmpdir).

Deployed container with debian-stretch@0.1.5 being latest verson.

Created debian-stretch@0.1.6 and uploaded.

Deployed new container with debian-stretch@0.1.6.

New template gets downloaded:

```
athen243:/var/snap/subutai-master/common/lxc# ls -l tmpdir/
total 168264
-rw-r--r-- 1 root root 86151596 Jan 26 01:47 debian-stretch-subutai-template_0.1.5_amd64.tar.gz
-rw-r--r-- 1 root root 86144587 Jan 26 02:35 debian-stretch-subutai-template_0.1.6_amd64.tar.gz
```

But it doesn't look as if it being used:

```
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

```
athen243:/var/snap/subutai-master/common/lxc# cat Container-1-Nyz-xlr-15-3/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

**Comment by Dilshat** [comment]
@lbthomsen did you update file `/etc/subutai-template-version` in version 16 of the template?

**Comment by Dilshat** [comment]
can you do `cat /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/config` ?

**Comment by Dilshat** [comment]
also please make sure you update RH. Agent was supposed to remove old template archive but it is still present for some reason.

**Comment by lbthomsen** [comment]
/etc/subutai-template-version gets generated with the template so will always be right.  I only introduced that around 0.1.4 or so.

At the time of writing peer was a master peer and I did check subutai update rh.

And this is interesting because:

```
# Subutai debian-stretch template                                                                                                                                                                                    
lxc.arch = amd64                                                                                                                                                                                                     
# Network configuration                                                                                                                                                                                              
lxc.network.type = veth                                                                                                                                                                                              
lxc.network.flags = up                                                                                                                                                                                               
lxc.network.hwaddr = 00:16:3e:47:49:41                                                                                                                                                                               
subutai.config.path = /snap/subutai-master/current/etc                                                                                                                                                               
subutai.parent = debian-stretch                                                                                                                                                                                      
lxc.mount.entry = /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/home home none bind,rw 0 0                                                                                                            
lxc.mount.entry = /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/opt opt none bind,rw 0 0                                                                                                              
lxc.mount.entry = /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/var var none bind,rw 0 0                                                                                                              
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz                                                                                                           
subutai.template.version = 0.1.6                                                                                                                                                                                     
lxc.id_map = u 0 1507328 65536                                                                                                                                                                                       
lxc.id_map = g 0 1507328 65536                                                                                                                                                                                       
                                                                                                                                                                                                                     
# systemD configuration                                                                                                                                                                                              
lxc.mount.auto = cgroup:mixed                                                                                                                                                                                        
lxc.autodev = 1                                                                                                                                                                                                      
lxc.kmsg = 0                                                                                                                                                                                                         
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf                                                                                                                                       
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs                                                                                                                                         
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface                                                                                                                                        
                                                                                                                                                                                                                     
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf                                                                                                                                       
lxc.rootfs = /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/rootfs                                                                                                                                     
lxc.rootfs.backend = btrfs                                                                                                                                                                                           
lxc.utsname = Container-1-Nyz-xlr-15-3                                                                                                                                                                               
lxc.mount = /var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3/fstab                                                                                                                                       
lxc.network.veth.pair = 00163e474941                                                                                                                                                                                 
lxc.network.mtu = 1300                                                                                                                                                                                               
lxc.network.ipv4 = 172.17.96.3/24                                                                                                                                                                                    
lxc.network.ipv4.gateway = 172.17.96.252                                                                                                                                                                             
#vlan_id = 15                                                                                                                                                                                                        
lxc.console = none                                                                                                                                                                                                   
lxc.tty = 0                                                                                                                                                                                                          
lxc.cgroup.devices.deny = c 5:1 rwm
athen243:/var/snap/subutai-master/common/lxc/Container-1-Nyz-xlr-15-3# cat rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

BUT:

```
athen243:/var/snap/subutai-master/common/lxc# ls -ls tmpdir/
total 249108
84136 -rw-r--r-- 1 root root 86151596 Jan 26 01:47 debian-stretch-subutai-template_0.1.5_amd64.tar.gz
84128 -rw-r--r-- 1 root root 86144587 Jan 26 02:35 debian-stretch-subutai-template_0.1.6_amd64.tar.gz
80844 -rw-r--r-- 1 root root 82783312 Jan 26 03:06 ubuntu-xenial-subutai-template_0.1.6_amd64.tar.gz
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/config
# Subutai debian-stretch template                                                                                                                                                                                    
lxc.arch = amd64                                                                                                                                                                                                     
# Network configuration                                                                                                                                                                                              
lxc.network.type = veth                                                                                                                                                                                              
lxc.network.flags = up                                                                                                                                                                                               
lxc.network.hwaddr = 00:16:3e:c2:4d:fa                                                                                                                                                                               
subutai.config.path = /snap/subutai-master/current/etc                                                                                                                                                               
subutai.parent = debian-stretch                                                                                                                                                                                      
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/home home none bind,rw 0 0                                                                                                                      
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/opt opt none bind,rw 0 0                                                                                                                        
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/var var none bind,rw 0 0                                                                                                                        
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz                                                                                                           
subutai.template.version = 0.1.6                                                                                                                                                                                     
lxc.rootfs = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs                                                                                                                                               
lxc.utsname = debian-stretch                                                                                                                                                                                         
lxc.mount = /var/snap/subutai-master/common/lxc/debian-stretch/fstab                                                                                                                                                 
lxc.id_map = u 0 1114112 65536                                                                                                                                                                                       
lxc.id_map = g 0 1114112 65536                                                                                                                                                                                       
                                                                                                                                                                                                                     
# systemD configuration                                                                                                                                                                                              
lxc.mount.auto = cgroup:mixed                                                                                                                                                                                        
lxc.autodev = 1                                                                                                                                                                                                      
lxc.kmsg = 0                                                                                                                                                                                                         
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf                                                                                                                                       
lxc.network.link = lxcbr0                                                                                                                                                                                            
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs                                                                                                                                         
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface                                                                                                                                        
                                                                                                                                                                                                                     
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf                                                                                                                                       
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5                                                                                                                                                                                                 
athen243:/var/snap/subutai-master/common/lxc# subutai destroy debian-stretch                                                                                                                                         
INFO[2018-01-26 07:48:33] debian-stretch is destroyed                                                                                                                                                                
athen243:/var/snap/subutai-master/common/lxc# subutai import debian-stretch                                                                                                                                          
INFO[2018-01-26 07:48:41] Importing debian-stretch                                                                                                                                                                   
INFO[2018-01-26 07:48:41] Version: 0.1.6                                                                                                                                                                             
INFO[2018-01-26 07:48:41] Template's owner signature verified                                                                                                                                                        
INFO[2018-01-26 07:48:43] Downloading debian-stretch                                                                                                                                                                 
 82.15 MiB / 82.15 MiB [================================================================================================================================================================================] 100.00% 11s
INFO[2018-01-26 07:48:55] File integrity verified                                                                                                                                                                    
INFO[2018-01-26 07:48:55] Unpacking template debian-stretch                                                                                                                                                          
INFO[2018-01-26 07:48:57] Installing template debian-stretch           
athen243:/var/snap/subutai-master/common/lxc# ls -ls tmpdir/
total 249108
84136 -rw-r--r-- 1 root root 86151596 Jan 26 01:47 debian-stretch-subutai-template_0.1.5_amd64.tar.gz
84128 -rw-r--r-- 1 root root 86144587 Jan 26 07:48 debian-stretch-subutai-template_0.1.6_amd64.tar.gz
80844 -rw-r--r-- 1 root root 82783312 Jan 26 03:06 ubuntu-xenial-subutai-template_0.1.6_amd64.tar.gz
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/config
# Subutai debian-stretch template
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:c2:4d:fa
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz
subutai.template.version = 0.1.6
lxc.rootfs = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.utsname = debian-stretch
lxc.mount = /var/snap/subutai-master/common/lxc/debian-stretch/fstab
lxc.id_map = u 0 1114112 65536
lxc.id_map = g 0 1114112 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.network.link = lxcbr0
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.6
athen243:/var/snap/subutai-master/common/lxc# 
```

Something is quite odd.

**Comment by lbthomsen** [comment]
There was a new rh just now - let me try this again.

**Comment by lbthomsen** [comment]
Going through the steps:

1. debian-stretch@0.1.5 latest version in Kurjub

```
athen243:/var/snap/subutai-master/common/lxc# subutai destroy debian-stretch
INFO[2018-01-26 07:56:23] debian-stretch is destroyed                  
athen243:/var/snap/subutai-master/common/lxc# ls -l tmpdir/
total 0
```

Clean!

2. Deploy debian-stretch env from masterhub - result:

```
athen243:/var/snap/subutai-master/common/lxc# ls -l tmpdir/
total 84136                                                                                                                                                                                                          
-rw-r--r-- 1 root root 86151596 Jan 26 07:59 debian-stretch-subutai-template_0.1.5_amd64.tar.gz                                                                                                                      
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/config                                                                                                                                              
# Subutai debian-stretch template                                                                                                                                                                                    
lxc.arch = amd64                                                                                                                                                                                                     
# Network configuration                                                                                                                                                                                              
lxc.network.type = veth                                                                                                                                                                                              
lxc.network.flags = up                                                                                                                                                                                               
lxc.network.hwaddr = 00:16:3e:c2:4d:fa                                                                                                                                                                               
subutai.config.path = /snap/subutai-master/current/etc                                                                                                                                                               
subutai.parent = debian-stretch                                                                                                                                                                                      
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/home home none bind,rw 0 0                                                                                                                      
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/opt opt none bind,rw 0 0                                                                                                                        
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/var var none bind,rw 0 0                                                                                                                        
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.5_amd64.tar.gz                                                                                                           
subutai.template.version = 0.1.5                                                                                                                                                                                     
lxc.rootfs = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs                                                                                                                                               
lxc.utsname = debian-stretch                                                                                                                                                                                         
lxc.mount = /var/snap/subutai-master/common/lxc/debian-stretch/fstab                                                                                                                                                 
lxc.id_map = u 0 1114112 65536                                                                                                                                                                                       
lxc.id_map = g 0 1114112 65536                                                                                                                                                                                       
                                                                                                                                                                                                                     
# systemD configuration                                                                                                                                                                                              
lxc.mount.auto = cgroup:mixed                                                                                                                                                                                        
lxc.autodev = 1                                                                                                                                                                                                      
lxc.kmsg = 0                                                                                                                                                                                                         
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf                                                                                                                                       
lxc.network.link = lxcbr0                                                                                                                                                                                            
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs                                                                                                                                         
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface                                                                                                                                        
                                                                                                                                                                                                                     
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf                                                                                                                                       
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5                                                                                                                                                                                                 
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.5-aGn-dma-8-3/config                                                                                                                                   
# Subutai debian-stretch template                                                                                                                                                                                    
lxc.arch = amd64                                                                                                                                                                                                     
# Network configuration                                                                                                                                                                                              
lxc.network.type = veth                                                                                                                                                                                              
lxc.network.flags = up                                                                                                                                                                                               
lxc.network.hwaddr = 00:16:3e:96:89:b4                                                                                                                                                                               
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.5-aGn-dma-8-3/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.5-aGn-dma-8-3/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.5-aGn-dma-8-3/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.5_amd64.tar.gz
subutai.template.version = 0.1.5
lxc.id_map = u 0 196608 65536
lxc.id_map = g 0 196608 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
lxc.rootfs = /var/snap/subutai-master/common/lxc/stretch-0.1.5-aGn-dma-8-3/rootfs
lxc.rootfs.backend = btrfs
lxc.utsname = stretch-0.1.5-aGn-dma-8-3
lxc.mount = /var/snap/subutai-master/common/lxc/stretch-0.1.5-aGn-dma-8-3/fstab
lxc.network.veth.pair = 00163e9689b4
lxc.network.mtu = 1300
lxc.network.ipv4 = 172.21.47.3/24
lxc.network.ipv4.gateway = 172.21.47.252
#vlan_id = 8
lxc.console = none
lxc.tty = 0
lxc.cgroup.devices.deny = c 5:1 rwm
lxc.cgroup.memory.limit_in_bytes = 512M
lxc.cgroup.cpu.cfs_quota_us = 100000
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.5-aGn-dma-8-3/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

So far so good.

3. Uploading debian-stretch@0.1.6

4. Deploy env from masterhub based on 0.1.6 - result:

```
athen243:/var/snap/subutai-master/common/lxc# subutai update rh
INFO[2018-01-26 08:10:30] No update is available
athen243:/var/snap/subutai-master/common/lxc# ls -ls tmpdir/
total 168264
84136 -rw-r--r-- 1 root root 86151596 Jan 26 07:59 debian-stretch-subutai-template_0.1.5_amd64.tar.gz
84128 -rw-r--r-- 1 root root 86144587 Jan 26 08:08 debian-stretch-subutai-template_0.1.6_amd64.tar.gz
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/config 
# Subutai debian-stretch template
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:c2:4d:fa
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz
subutai.template.version = 0.1.6
lxc.rootfs = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.utsname = debian-stretch
lxc.mount = /var/snap/subutai-master/common/lxc/debian-stretch/fstab
lxc.id_map = u 0 1114112 65536
lxc.id_map = g 0 1114112 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.network.link = lxcbr0
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.6-E7i-ubi-14-3/config 
# Subutai debian-stretch template
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:dd:34:36
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-E7i-ubi-14-3/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-E7i-ubi-14-3/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-E7i-ubi-14-3/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz
subutai.template.version = 0.1.6
lxc.id_map = u 0 1572864 65536
lxc.id_map = g 0 1572864 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
lxc.rootfs = /var/snap/subutai-master/common/lxc/stretch-0.1.6-E7i-ubi-14-3/rootfs
lxc.rootfs.backend = btrfs
lxc.utsname = stretch-0.1.6-E7i-ubi-14-3
lxc.mount = /var/snap/subutai-master/common/lxc/stretch-0.1.6-E7i-ubi-14-3/fstab
lxc.network.veth.pair = 00163edd3436
lxc.network.mtu = 1300
lxc.network.ipv4 = 172.26.138.3/24
lxc.network.ipv4.gateway = 172.26.138.252
#vlan_id = 14
lxc.console = none
lxc.tty = 0
lxc.cgroup.devices.deny = c 5:1 rwm
lxc.cgroup.memory.limit_in_bytes = 512M
lxc.cgroup.cpu.cfs_quota_us = 100000
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.6-E7i-ubi-14-3/rootfs/etc/subutai_template_version 
debian-stretch@0.1.5
```

It appears that:

1. tmpdir does not get cleaned
2. Somehow oddly the config from the right template is used but the rootfs at least is from a wrong one.

If I manually clear and deploy new:

```
athen243:/var/snap/subutai-master/common/lxc# rm tmpdir/*
athen243:/var/snap/subutai-master/common/lxc# subutai destroy debian-stretch
runtime/cgo: pthread_create failed: Resource temporarily unavailable
INFO[2018-01-26 08:12:55] debian-stretch is destroyed
```

deploying env

```
athen243:/var/snap/subutai-master/common/lxc# ls -l tmpdir/
total 84128
-rw-r--r-- 1 root root 86144587 Jan 26 08:15 debian-stretch-subutai-template_0.1.6_amd64.tar.gz
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/config 
# Subutai debian-stretch template
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:c2:4d:fa
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/debian-stretch/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz
subutai.template.version = 0.1.6
lxc.rootfs = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.utsname = debian-stretch
lxc.mount = /var/snap/subutai-master/common/lxc/debian-stretch/fstab
lxc.id_map = u 0 1114112 65536
lxc.id_map = g 0 1114112 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.network.link = lxcbr0
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
athen243:/var/snap/subutai-master/common/lxc# cat debian-stretch/rootfs/etc/subutai_template_version 
debian-stretch@0.1.6
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.6-SCl-ytr-8-3/config 
# Subutai debian-stretch template
lxc.arch = amd64
# Network configuration
lxc.network.type = veth
lxc.network.flags = up
lxc.network.hwaddr = 00:16:3e:30:68:cf
subutai.config.path = /snap/subutai-master/current/etc
subutai.parent = debian-stretch
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-SCl-ytr-8-3/home home none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-SCl-ytr-8-3/opt opt none bind,rw 0 0
lxc.mount.entry = /var/snap/subutai-master/common/lxc/stretch-0.1.6-SCl-ytr-8-3/var var none bind,rw 0 0
subutai.template.package = /mnt/lib/lxc/lxc-data/tmpdir/debian-stretch-subutai-template_0.1.6_amd64.tar.gz
subutai.template.version = 0.1.6
lxc.id_map = u 0 196608 65536
lxc.id_map = g 0 196608 65536

# systemD configuration
lxc.mount.auto = cgroup:mixed
lxc.autodev = 1
lxc.kmsg = 0
lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.common.conf
lxc.rootfs.mount = /var/snap/subutai-master/common/lxc/debian-stretch/rootfs
lxc.network.script.up = /snap/subutai-master/current/bin/create_ovs_interface

lxc.include = /snap/subutai-master/current/share/lxc/config/ubuntu.userns.conf
lxc.rootfs = /var/snap/subutai-master/common/lxc/stretch-0.1.6-SCl-ytr-8-3/rootfs
lxc.rootfs.backend = btrfs
lxc.utsname = stretch-0.1.6-SCl-ytr-8-3
lxc.mount = /var/snap/subutai-master/common/lxc/stretch-0.1.6-SCl-ytr-8-3/fstab
lxc.network.veth.pair = 00163e3068cf
lxc.network.mtu = 1300
lxc.network.ipv4 = 172.20.12.3/24
lxc.network.ipv4.gateway = 172.20.12.252
#vlan_id = 8
lxc.console = none
lxc.tty = 0
lxc.cgroup.devices.deny = c 5:1 rwm
lxc.cgroup.memory.limit_in_bytes = 512M
lxc.cgroup.cpu.cfs_quota_us = 100000
athen243:/var/snap/subutai-master/common/lxc# cat stretch-0.1.6-SCl-ytr-8-3/rootfs/etc/subutai_template_version 
debian-stretch@0.1.6
```

**Comment by Dilshat** [comment]
@lbthomsen can you repeat the experiment with templates created from original parents such as *master, ubuntu16* etc  to exclude the possibility of template creation specifics ?

**Comment by lbthomsen** [comment]
How?  I can't control the version of the original ones not can I reconstruct them as I don't know how they were made.

**Comment by Dilshat** [comment]
Fixed in 6.4.2
